src/main/java/macrobase/analysis/stats/mixture/FiniteGMM.java

```diff
@@ -38,6 +38,10 @@ public void train(List<Datum> data) {
         VariationalInference.trainMeanField(this, data, mixingComponents, clusters);
     }
 
+    /**
+     * @param datum
+     * @return log probability density of the given datum
+     */
     @Override
```

Comment:
this algorithm does not care how you think about outliers. probability density of the given datum being 0.9 means that there is `0.9 * ds` chance of a datum appearing within `ds` volume of the space surrounding the metrics of the datum.
Basically this number has the same meaning as the one you get from KDE